### PR TITLE
fuzz: Script validation flags

### PR DESCRIPTION
**Pull request by maflcko**

### CODE REVIEW

**Comment by laanwj** [comment]
Code review utACK fab15ff70e8db26942db83a9aab7ebf974e8b6b2
Haven't tried what code paths are reached in a fuzzer.

**Comment by MarcoFalke** [comment]
Great! Thanks for looking at this. If you find new coverage, don't forget to add them to https://github.com/bitcoin-core/qa-assets/tree/master/fuzz_seed_corpus/script_flags, such that progress is accumulative over all machines that run the fuzzer and new fuzzers don't have to start from zero.

Related: #15295 (comes with instructions on how to build the fuzzer)

**Comment by practicalswift** [comment]
Strong concept ACK

Thanks for adding more fuzzing targets! Very much needed!

File: src/test/fuzz/script_flags.cpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+        unsigned int verify_flags;
+        ds >> verify_flags;
+
+        if (!IsValidFlagCombination(verify_flags)) return;
+
+        unsigned int fuzzed_flags;
+        ds >> fuzzed_flags;
+
+        for (unsigned i = 0; i < tx.vin.size(); ++i) {
```
Comment (practicalswift) [inline src/test/fuzz/script_flags.cpp]: The dropped `int` here intentional? Same result but looks non-idiomatic :-)

**Comment by jamesob** [comment]
Post-merge utACK https://github.com/MarcoFalke/bitcoin/commit/743c2f461c0e7c58b1e5c6ddfeaf4f0ff6e6ca85

Nice!

**Pull request merged**